src/vue/components/panel-compact.vue

```diff
@@ -124,59 +137,70 @@
-        <inflection-browser :data="data.inflectionBrowserData" :infl-browser-tables-collapsed="data.inflBrowserTablesCollapsed"
+           v-if="settings.contentOptions.items" v-show="$store.state.app.tabState.inflectionsbrowser">
+        <inflection-browser :data="data.inflectionBrowserData"
+                            :infl-browser-tables-collapsed="data.inflBrowserTablesCollapsed"
                             :language-id="inflectionBrowserLanguageID" @contentwidth="setContentWidth">
         </inflection-browser>
       </div>
       <div class="alpheios-panel__tab-panel alpheios-panel__tab__grammar
-            alpheios-panel__tab-panel--no-padding alpheios-panel__tab-panel--fw" v-show="data.tabs.grammar">
-        <grammar :res="data.grammarRes"></grammar>
+            alpheios-panel__tab-panel--no-padding alpheios-panel__tab-panel--fw"
+           v-show="$store.state.app.tabState.grammar">
+        <grammar></grammar>
       </div>
-      <div class="alpheios-panel__tab-panel alpheios-panel__tab__treebank
-            alpheios-panel__tab-panel--no-padding alpheios-panel__tab-panel--fw" v-if="data.treebankComponentData && data.settings"
-           v-show="treebankTabVisible">
-        <treebank :locale="data.settings.locale.currentValue" :res="data.treebankComponentData.data"
-                  :visible="data.treebankComponentData.visible" @treebankcontentwidth="setTreebankContentWidth">
-        </treebank>
+      <div
+          class="alpheios-panel__tab-panel alpheios-panel__tab__treebank alpheios-panel__tab-panel--no-padding alpheios-panel__tab-panel--fw"
+          v-if="$store.getters['app/hasTreebankData']" v-show="$store.state.app.tabState.treebank">
+        <treebank @treebankcontentwidth="setTreebankContentWidth"></treebank>
       </div>
-      <div class="alpheios-panel__tab-panel alpheios-panel__tab__status" v-show="data.tabs.status">
+      <div class="alpheios-panel__tab-panel alpheios-panel__tab__status" v-show="$store.state.app.tabState.status">
         <!-- Messages to be displayed in a status panel -->
         <div v-for="message in data.messages">
           <div class="alpheios-panel__message">{{message}}</div>
         </div>
       </div>
-      <div class="alpheios-panel__tab-panel alpheios-panel__tab__status" v-show="data.tabs.user">
+      <div class="alpheios-panel__tab-panel alpheios-panel__tab__status" v-show="$store.state.app.tabState.user">
         <user-auth :auth="data.auth"></user-auth>
       </div>
-      <div class="alpheios-panel__tab-panel alpheios-panel__tab__options" v-show="data.tabs.options">
+      <div class="alpheios-panel__tab-panel alpheios-panel__tab__word-usage"
+           v-show="$store.state.app.tabState.wordUsage" v-if="$store.getters['app/hasWordUsageExamplesData']">
+        <word-usage-examples-block
+            :wordUsageList="$store.state.app.wordUsageExamplesData.wordUsageExamples"
```

Comment:
I agree, just did not have time to refactor this part (and also because PR already have so many changes I wanted to keep it at least manageable). Pulling this data from store/controller would make much more sense. I will update the code.

Reply:
ok, it's also okay to leave it as is for this PR but please mark with with a TODO comment

Reply:
OK, will do that